### ISSUE TITLE
navigationBar.update() not working as intended

### ISSUE THREAD

**Issue by sunny061** [AppGyver/steroids#981]
I want to override the automatic back button in the navbar which is created after I push a view.

I have this navbar in my view that I want to overwrite.
![bug3](https://cloud.githubusercontent.com/assets/8417115/8448665/b552dc36-1ffc-11e5-92d8-70467f5ccf0d.png)

So I wrote the code below in my controller.
![bug1](https://cloud.githubusercontent.com/assets/8417115/8448626/3accc792-1ffc-11e5-8b0d-01e29589fda3.png)

But the result I am getting is this...
![bug 2](https://cloud.githubusercontent.com/assets/8417115/8448627/3ad433e2-1ffc-11e5-8289-1a5a41980920.png)

Thanks in advance for the help!

**Comment by sunny061** [comment]
Update:
It turns out its a timing issue. If I put a setTimeout before updating the navbar, then I am able to successfully update the navbar. Although it doesn't look nice when the navbar suddenly changes. Is there a way to change it instantly as the view is pushed?

**Comment by videege** [comment]
+1.  I am also experiencing a timing issue where my changes to the navbar are briefly present and then wiped out; if I delay my changes via a timeout then they work just fine.

**Comment by harsaharsa** [comment]
This is because the `<super-navbar>` web component internally calls the same `supersonic.ui.navigationBar.update` API after it has loaded. If your controller code executes before the web components have loaded, this cause the error.

Putting your code inside a [WebComponentsReady event listener](https://github.com/webcomponents/webcomponentsjs#webcomponentsready) might fix this. We are looking into ways to make native UI initialization smoother, stay tuned!

**Comment by sunny061** [comment]
Tested the WebComponentsReady event listener method and it did not produce the desired result. So I resorted back to updating my navbar after the supersonic.ui.views.current.whenVisible() callback fires until I can find a safer/smoother workaround.

**Comment by romecode** [comment]
@sunny061 
Does your onTap function work? I wrote the same exact code just to test and I get nothing.

**Comment by sunny061** [comment]
@romecode yes, the onTap is working as long as the navbar is updated after the view is fully loaded.

**Comment by romecode** [comment]
supersonic.ui.views.current.whenVisible(function(){
			 supersonic.ui.navigationBar.update(opt);
});

Like so? I cant seem to get it to fire.. strange

**Comment by sunny061** [comment]
@romecode Sorry I forgot to mention that. I also ended up doing everything related to Navbars using API calls including creating them.

**Comment by zeitrapha** [comment]
may it be the reason, why i am getting this scenario?
http://community.appgyver.com/t/configured-back-button-replaced
